src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,581 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
 
-            /* huisnummer in straatnaam */
-            if ((string.IsNullOrEmpty(houseNumber) || houseNumber0 == "00000000") && string.IsNullOrEmpty(index) && !string.IsNullOrEmpty(streetName))
+            // huisnummer in straatnaam
+            if ((string.IsNullOrEmpty(houseNumber) || formattedHouseNumber == "00000000") &&
+                string.IsNullOrEmpty(index) &&
+                !string.IsNullOrEmpty(streetName))
             {
-                var requestStraatnaamWithoutStraatnaam = streetName.Replace(StripStreetName(streetName), "").Replace(",", string.Empty).Replace(" ", string.Empty);
-                //huisnummer als prefix in straatnaam
+                int hnr;
+
+                var requestStraatnaamWithoutStraatnaam = streetName
+                    .Replace(StripStreetName(streetName), string.Empty)
+                    .Replace(",", string.Empty)
+                    .Replace(" ", string.Empty);
+
+                var possibleNumberInStreet = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
+                var possibleIndexInStreet = FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam));
+
+                // huisnummer als prefix in straatnaam
+                // TODO: [0-9]+ ?
                 if (new Regex("[0-9] ").IsMatch(streetName))
                 {
-                    if (FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)) == "0000"
-                        && FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
+                    if (possibleIndexInStreet == "0000"
+                        && possibleNumberInStreet != null
+                        && int.TryParse(possibleNumberInStreet, out hnr)
                         && hnr != 0)
                     {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
+                        formattedHouseNumber = possibleNumberInStreet;
                     }
                 }
-                //huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+
+                // huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+                // TODO: [0-9]+ ?
                 if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))
                 {
-                    if (FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
+                    if (possibleNumberInStreet != null
+                        && int.TryParse(possibleNumberInStreet, out hnr)
                         && hnr != 0
+                        // TODO: [0-9]+ ?
                         && new Regex("[0-9][a-zA-Z]$").IsMatch(requestStraatnaamWithoutStraatnaam))
```

Comment:
Shouldn't this be `[0-9]+` instead?

Reply:
In retrospect, this will match "XA " so "blablaXXA " is seen as a match too :)

It would still be cleaner with `[0-9]+[a-zA-Z]` explicitly.